core/sql/executor/ExHdfsScan.cpp

```diff
@@ -367,6 +367,9 @@ ExWorkProcRetcode ExHdfsScanTcb::work()
   char cursorId[8];
   HdfsFileInfo *hdfo = NULL;
   Lng32 openType = 0;
+  int hiveScanMode = 0;
+
+  hiveScanMode = CmpCommon::getDefaultLong(HIVE_SCAN_SPECIAL_MODE);
```

Comment:
thanks Selva, this explains my puzzle. In the hive regress test, I have to 
cqd CALL_EMBEDDED_ARKCMP 'ON';
It must be due to the reason you mentioned, here, I will add this as a TDB flag.